python_benchmarks/test_softmax_fwd.py

```diff
@@ -63,8 +67,36 @@ def test_softmax_fwd_benchmark(
         softmax_fwd_fusion(fd, torch_dtype_to_nvfuser_dtype(dtype), reduction_axis)
 
     if not disable_validation:
-        eager_output = torch.nn.functional.softmax(inputs[0], dim=reduction_axis)
+        eager_output = softmax_fwd_fn([inputs[0], reduction_axis])
         fd.validate(inputs, [eager_output])
 
     if not disable_benchmarking:
         run_benchmark(benchmark, fd.execute, inputs)
+
+
+@pytest.mark.parametrize("size", generate_input_sizes(dims=2))
+@pytest.mark.parametrize("dtype", FLOAT_DTYPES)
+@pytest.mark.parametrize("reduction_axis", [0, 1])
+def test_softmax_fwd_eager_benchmark(
+    benchmark,
+    size: tuple,
+    dtype: torch.dtype,
+    reduction_axis: int,
+):
+    clear_cuda_cache()
+    input = torch.randn(*size, device="cuda", dtype=dtype)
+    run_benchmark(benchmark, softmax_fwd_fn, [input, reduction_axis])
+
+
+@pytest.mark.parametrize("size", generate_input_sizes(dims=2))
+@pytest.mark.parametrize("dtype", FLOAT_DTYPES)
+@pytest.mark.parametrize("reduction_axis", [0, 1])
+def test_softmax_fwd_compile_benchmark(
+    benchmark,
+    size: tuple,
+    dtype: torch.dtype,
+    reduction_axis: int,
+):
+    clear_cuda_cache()
+    input = torch.randn(*size, device="cuda", dtype=dtype)
+    run_benchmark(benchmark, torch.compile(softmax_fwd_fn), [input, reduction_axis])
```

Comment:
Feel free to ignore, but since these tests are nearly identical is it practical to parametrize a single test here over whether `torch.compile` is used or not? I am thinking something like
```python
@pytest.mark.parametrize("size", generate_input_sizes(dims=2))
@pytest.mark.parametrize("dtype", FLOAT_DTYPES)
@pytest.mark.parametrize("reduction_axis", [0, 1])
@pytest.mark.parametrize("compile", [False, True], ids=["eager", "compile"])
def test_softmax_fwd_compile_benchmark(
    benchmark,
    size: tuple,
    dtype: torch.dtype,
    reduction_axis: int,
    compile: bool,
):
    clear_cuda_cache()
    input = torch.randn(*size, device="cuda", dtype=dtype)
    run_benchmark(
        benchmark,
        torch.compile(softmax_fwd_fn) if compile else softmax_fwd_fn,
        [input, reduction_axis])
```

Reply:
Same for bwd tests above.

Reply:
I considered this. The main reason I split them is to be able to run them independently using filtering flags (`-k eager, compile`). I am looking into if I can filter on the parametrization flag in a clean way.
The other option could be to merge and use flags similar to `disable-validation` and `disable-benchmarking.`

Reply:
You can use `-k` with the `ids` that print I believe. In this case the test prints like `test_softmax_fwd_compile_benchmark[eager-0-torch.float32-(16, 768)]`. I guess you'd want to rename the function to `test_softmax_fwd_benchmark` instead, but point being that the `ids` argument to `parametrize` helps to make these filterable.

Reply:
On that topic, I actually had some trouble filtering on shapes because  of how they print with commas. For example try filtering down to only tests that run with size `(16, 6400)`. I still don't know how to use `-k` to do that without modifying the ids. In cases like that, you can use something like this though
```python
@pytest.mark.parametrize("size", generate_input_sizes(dims=2), ids=lambda s: '_'.join(str(si) for si in s))
def test_foo(
    ...
```
This will print names like `python_benchmarks/test_softmax_fwd.py::test_foo[16_6400]` which you can filter to using `pytest foo.py -k "foo and 16_6400]"`.

Reply:
It is filterable on ids. But I am having to filter the entire test ID: `-k test_softmax_fwd_compile_benchmark[eager-0-torch.float32-(16, 768)]` instead of something like `-k eager`. Although, this can work too (`test_softmax_fwd_compile_benchmark[*eager*]`), my only reluctance is that it is quite verbose and test name dependent. I believe I can simplify it by adding parser options.

Reply:
> On that topic, I actually had some trouble filtering on shapes because of how they print with commas. For example try filtering down to only tests that run with size `(16, 6400)`. I still don't know how to use `-k` to do that without modifying the ids. In cases like that, you can use something like this though
> 
> ```python
> @pytest.mark.parametrize("size", generate_input_sizes(dims=2), ids=lambda s: '_'.join(str(si) for si in s))
> def test_foo(
>     ...
> ```
> 
> This will print names like `python_benchmarks/test_softmax_fwd.py::test_foo[16_6400]` which you can filter to using `pytest foo.py -k "foo and 16_6400]"`.

That's a good point, let me see if I can change this without needing significant changes in the CI. @xwang233 has scripts to process the outputs generated by pytest and assumes the sizes to be integers at the moment.